.github/workflows/release.yml

```diff
@@ -0,0 +1,55 @@
+name: release
+on:
+  push:
+    branches: [master, main, 'andrei/workflows/ruby-downloader']
+    tags: ["*"]
+    paths-ignore: ['**.md']
+  workflow_dispatch: #allows manual trigger
+
+concurrency: production
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+      - name: Get next release version (dry run)
+        id: taggerDryRun
+        uses: anothrNick/github-tag-action@1.61.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WITH_V: true
+          DRY_RUN: true
+      - name: echo new tag
+        run: |
+          export NEXT_VERSION=${{ steps.taggerDryRun.outputs.new_tag }}
+          echo "The next tag version will be: $NEXT_VERSION"
+      - name: echo tag
+        run: |
+          echo "The current tag is: ${{ steps.taggerDryRun.outputs.tag }}"
+      - name: Compile
+        run: sbt clean stage
+      - name: Generate Ruby Types
+        run: ./standalone --withLanguageFrontend RUBYSRC 
+      - name: Checksum
+        run: sha512sum src/main/resources/ruby/builtin_types.zip > src/main/resources/ruby/builtin_types.zip.sha512
```

Comment:
Maybe you could make `standalone` generate to the current working directory to simplify this calculation, or some `-o` output directory argument.